### PR TITLE
Updated dockerignore from the root

### PR DESCRIPTION
**Pull request by Tansito**
### Summary

This PR is related with #424 and #380. 

After #424 I didn't notice a lot of improvements in the time spent building the applications and I think the reason is because I was testing the build with `docker-compose`. This scenario uses the root where we had a `.dockerignore` without being updated. Now the `chown` step almost takes 30 seconds.

I think this is the most used use-case right now.

### Details and comments

- Updated `.dockerignore` from the root.
- Updated `.dockerignore` from `repository` to include more exceptions
- The most important changes were: `web/media and web/static` in our case are only `media/ & static/`

### CODE REVIEW

**Comment by psschwei** [comment]
> I was testing the build with `docker-compose`. This scenario uses the root where we had a `.dockerignore` without being updated. Now the `chown` step almost takes 30 seconds.
> 
> I think this is the most used use-case right now.

Noting here to document this as part of #372  ... I think I've been building using the Makefile and then running `docker compose up`...

**Comment by Tansito** [comment]
Mmm... Good question @psschwei to be honest the last days I was doing:
- `docker compose build`
- `docker compose up`

But I didn't use `docker-compose` too much so probably I'm not the best example. What do you usually do, @IceKhan13 ?

**Review by psschwei — COMMENTED** [review]
Using `docker-compose -f docker-compose-dev.yml build` failed for me with this error:

```
 => [2/5] COPY --chown=: ./client ./qs                                                                                                                                                  3.6s
failed to solve: failed to compute cache key: failed to calculate checksum of ref moby::my93we75g76b23zv5lqvupuuj: "/docs/guides": not found
```

(hard to tell which build is failing, need to disable buildkit and run again...)

**Comment by Tansito** [comment]
I recommend you to try with #399 @psschwei , I fixed some problems in that PR related with the build process and configuration.

**Comment by Tansito** [comment]
Oh well, yeah I was thinking for `development` only. Like:

```
docker-compose -f docker-compose-dev.yml build
```

The build step for `docker-compose.yml` has no sense because as you mentioned there is no image to build.

File: .dockerignore [hunk 1]
```diff
@@ -1,14 +1,50 @@
...
+*.css.map
+*.egg-info
+*.sql.gz
+.cache
+.idea/workspace.xml
+.project
+.pydevproject
+__pycache__
+dist
+docs
```
Comment (psschwei) [inline .dockerignore:39]: I think this is what was causing my error

Comment (Tansito) [reply .dockerignore:39]: Oh true, because we are using them so we can't remove them from the build. Good catch! 💯

Comment (psschwei) [reply .dockerignore:39]: well, that was part of it... failed again on `requirements.txt` ...

let me try the other PR like you said I should :smile:

**Comment by psschwei** [comment]
Images built fine, but the notebook fails when trying to create the gateway provider:

```
qs-jupyter                       | [W 2023-04-12 16:43:35.812 ServerApp] Notebook 01_intro_level_1.ipynb is not trusted
keycloak                         | 2023-04-12 16:43:37,459 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: Base URL: <unset>, Hostname: <request>, Strict HTTPS: false, Path: <request>, Strict BackChannel: false, Admin URL: <unset>, Admin: <request>, Port: -1, Proxied: true
keycloak                         | 2023-04-12 16:43:38,385 WARN  [io.quarkus.agroal.runtime.DataSources] (main) Datasource <default> enables XA but transaction recovery is not enabled. Please enable transaction recovery by setting quarkus.transaction-manager.enable-recovery=true, otherwise data may be lost if the application is terminated abruptly
quantum-serverless-postgres-1    | 2023-04-12 16:43:38.544 UTC [87] FATAL:  no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
quantum-serverless-postgres-1    | 2023-04-12 16:43:38.551 UTC [88] FATAL:  no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         | 2023-04-12 16:43:38,552 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': FATAL: no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         | 2023-04-12 16:43:38,553 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread: keycloak-default) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: FATAL: no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         |      at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:659)
keycloak                         |      at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:180)
keycloak                         |      at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
keycloak                         |      at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
keycloak                         |      at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
keycloak                         |      at org.postgresql.Driver.makeConnection(Driver.java:434)
keycloak                         |      at org.postgresql.Driver.connect(Driver.java:291)
keycloak                         |      at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:681)
keycloak                         |      at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:229)
keycloak                         |      at org.postgresql.ds.common.BaseDataSource.getConnection(BaseDataSource.java:103)
keycloak                         |      at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:49)
keycloak                         |      at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:35)
keycloak                         |      at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:232)
keycloak                         |      at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
keycloak                         |      at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
keycloak                         |      at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
keycloak                         |      at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
keycloak                         |      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
keycloak                         |      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
keycloak                         |      at java.base/java.lang.Thread.run(Thread.java:833)
keycloak                         |      Suppressed: org.postgresql.util.PSQLException: FATAL: no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         |              at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:659)
keycloak                         |              at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:180)
keycloak                         |              at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:244)
keycloak                         |              ... 17 more
keycloak                         | 
quantum-serverless-postgres-1    | 2023-04-12 16:43:38.943 UTC [89] FATAL:  no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
quantum-serverless-postgres-1    | 2023-04-12 16:43:38.945 UTC [90] FATAL:  no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         | 2023-04-12 16:43:38,945 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': FATAL: no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         | 2023-04-12 16:43:38,962 WARN  [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000569: Unable to persist Infinispan internal caches as no global state enabled
keycloak                         | 2023-04-12 16:43:38,966 WARN  [org.infinispan.PERSISTENCE] (keycloak-cache-init) ISPN000554: jboss-marshalling is deprecated and planned for removal
keycloak                         | 2023-04-12 16:43:38,985 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
keycloak                         | 2023-04-12 16:43:39,120 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000128: Infinispan version: Infinispan 'Triskaidekaphobia' 13.0.10.Final
keycloak                         | 2023-04-12 16:43:39,192 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000078: Starting JGroups channel `ISPN`
keycloak                         | 2023-04-12 16:43:39,192 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000088: Unable to use any JGroups configuration mechanisms provided in properties {}. Using default JGroups configuration!
keycloak                         | 2023-04-12 16:43:39,245 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
keycloak                         | 2023-04-12 16:43:39,245 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 20.00MB, but the OS only allocated 212.99KB
keycloak                         | 2023-04-12 16:43:39,246 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the send buffer of socket MulticastSocket was set to 1.00MB, but the OS only allocated 212.99KB
keycloak                         | 2023-04-12 16:43:39,246 WARN  [org.jgroups.protocols.UDP] (keycloak-cache-init) JGRP000015: the receive buffer of socket MulticastSocket was set to 25.00MB, but the OS only allocated 212.99KB
keycloak                         | 2023-04-12 16:43:41,252 INFO  [org.jgroups.protocols.pbcast.GMS] (keycloak-cache-init) 9c38e1ee591c-58828: no members discovered after 2002 ms: creating cluster as coordinator
keycloak                         | 2023-04-12 16:43:41,273 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000094: Received new cluster view for channel ISPN: [9c38e1ee591c-58828|0] (1) [9c38e1ee591c-58828]
keycloak                         | 2023-04-12 16:43:41,281 INFO  [org.infinispan.CLUSTER] (keycloak-cache-init) ISPN000079: Channel `ISPN` local address is `9c38e1ee591c-58828`, physical addresses are `[172.20.0.10:49947]`
keycloak                         | 2023-04-12 16:43:41,583 INFO  [org.infinispan.CLUSTER] (main) ISPN000080: Disconnecting JGroups channel `ISPN`
keycloak                         | 2023-04-12 16:43:41,710 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (development) mode
keycloak                         | 2023-04-12 16:43:41,710 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to obtain JDBC connection
keycloak                         | 2023-04-12 16:43:41,710 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: FATAL: no pg_hba.conf entry for host "172.20.0.10", user "testkeycloakuser", database "testkeycloakdb", no encryption
keycloak                         | 2023-04-12 16:43:41,711 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
keycloak exited with code 1
gateway                          | Internal Server Error: /dj-rest-auth/keycloak/login/
gateway                          | Traceback (most recent call last):
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 174, in _new_conn
gateway                          |     conn = connection.create_connection(
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/connection.py", line 72, in create_connection
gateway                          |     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
gateway                          |   File "/usr/local/lib/python3.9/socket.py", line 954, in getaddrinfo
gateway                          |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
gateway                          | socket.gaierror: [Errno -2] Name or service not known
gateway                          | 
gateway                          | During handling of the above exception, another exception occurred:
gateway                          | 
gateway                          | Traceback (most recent call last):
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
gateway                          |     httplib_response = self._make_request(
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 398, in _make_request
gateway                          |     conn.request(method, url, **httplib_request_kw)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 244, in request
gateway                          |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
gateway                          |   File "/usr/local/lib/python3.9/http/client.py", line 1285, in request
gateway                          |     self._send_request(method, url, body, headers, encode_chunked)
gateway                          |   File "/usr/local/lib/python3.9/http/client.py", line 1331, in _send_request
gateway                          |     self.endheaders(body, encode_chunked=encode_chunked)
gateway                          |   File "/usr/local/lib/python3.9/http/client.py", line 1280, in endheaders
gateway                          |     self._send_output(message_body, encode_chunked=encode_chunked)
gateway                          |   File "/usr/local/lib/python3.9/http/client.py", line 1040, in _send_output
gateway                          |     self.send(msg)
gateway                          |   File "/usr/local/lib/python3.9/http/client.py", line 980, in send
gateway                          |     self.connect()
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 205, in connect
gateway                          |     conn = self._new_conn()
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 186, in _new_conn
gateway                          |     raise NewConnectionError(
gateway                          | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f34807e2d00>: Failed to establish a new connection: [Errno -2] Name or service not known
gateway                          | 
gateway                          | During handling of the above exception, another exception occurred:
gateway                          | 
gateway                          | Traceback (most recent call last):
gateway                          |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 489, in send
gateway                          |     resp = conn.urlopen(
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 787, in urlopen
gateway                          |     retries = retries.increment(
gateway                          |   File "/usr/local/lib/python3.9/site-packages/urllib3/util/retry.py", line 592, in increment
gateway                          |     raise MaxRetryError(_pool, url, error or ResponseError(cause))
gateway                          | urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='keycloak', port=8080): Max retries exceeded with url: /realms/quantumserverless/protocol/openid-connect/token/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34807e2d00>: Failed to establish a new connection: [Errno -2] Name or service not known'))
gateway                          | 
gateway                          | During handling of the above exception, another exception occurred:
gateway                          | 
gateway                          | Traceback (most recent call last):
gateway                          |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 55, in inner
gateway                          |     response = get_response(request)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 197, in _get_response
gateway                          |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 56, in wrapper_view
gateway                          |     return view_func(*args, **kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/django/views/generic/base.py", line 104, in view
gateway                          |     return self.dispatch(request, *args, **kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
gateway                          |     response = self.handle_exception(exc)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
gateway                          |     self.raise_uncaught_exception(exc)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
gateway                          |     raise exc
gateway                          |   File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 506, in dispatch
gateway                          |     response = handler(request, *args, **kwargs)
gateway                          |   File "/usr/src/app/api/views.py", line 225, in post
gateway                          |     keycloak_response = requests.post(
gateway                          |   File "/usr/local/lib/python3.9/site-packages/requests/api.py", line 115, in post
gateway                          |     return request("post", url, data=data, json=json, **kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/requests/api.py", line 59, in request
gateway                          |     return session.request(method=method, url=url, **kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 587, in request
gateway                          |     resp = self.send(prep, **send_kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 701, in send
gateway                          |     r = adapter.send(request, **kwargs)
gateway                          |   File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 565, in send
gateway                          |     raise ConnectionError(e, request=request)
gateway                          | requests.exceptions.ConnectionError: HTTPConnectionPool(host='keycloak', port=8080): Max retries exceeded with url: /realms/quantumserverless/protocol/openid-connect/token/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f34807e2d00>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

**Review by psschwei — APPROVED** [review]

**Comment by psschwei** [comment]
All working now

**Pull request merged**